tests/test_unit/test_cdf_tk/test_commands/test_upload.py

```diff
@@ -142,3 +150,150 @@ def test_upload_raw_rows_dry_run(
             client.raw.rows.insert.assert_not_called()
             client.raw.databases.create.assert_not_called()
             client.raw.tables.create.assert_not_called()
+
+    def test_prepare_sorts_instance_selectors_by_dependencies(
```

Comment:
Replace all these tests with a single paramaterize tests.

Parameters should be, selectors_by_file, expected_selector_by_file.